### DISCUSSION TITLE
SBAS  problem: how to align TSX datasets by align_batch.csh?

### DISCUSSION THREAD

**Discussion by gaiyf** [gmtsar/gmtsar#551] [Q&A]
I want to process SBAS with TerraSAR-X data, but the **_align_batch.csh_** only have **ENVISAT, ALOS, ERS**. what should i do?
![Screenshot from 2022-11-14 16-21-49](https://user-images.githubusercontent.com/57431511/201611127-d69060e3-ead4-44ed-a92e-19f0e678b177.png)

**Comment by Xiaohua-Eric-Xu** [comment]
The current align_batch.csh script only works for the older satellites that needs focusing/re-focusing. For newer generation of satellites, you could align all images to a single master image by running `p2p_processing.csh` multiple times. I am thinking of writing up a script to do that but haven't got the time to do so.

**Reply by Xiaohua-Eric-Xu** [reply]
You'll need to use the config file to control the alignment, i.e. running step 1 and 2 only.

**Reply by gaiyf** [reply]
Thank you very much for your guidance and reply, and looking forward the update of GMTSAR.  The other thing that's i want to seek your advice is the **_inf.csh_** in  **_p2p_processing.csh_**  :   if i use the **_p2p_processing.csh_**  align  the images  **B**(_slave1_) to **A**(_master_)  and  **C**(_slave2_) to **A**(master) ,  can the **B** image  and **C** image make intferogram?

**Reply by Xiaohua-Eric-Xu** [reply]
Yes, you'll need to manipulate config file to do the following.
1) run step 1, 2 and 3 to align image B to A and produce topo_ra.grd using A as the master.
2) in another directory run step 1 and 2 to align C to A.
3) run step 4 5 6 to do interferometry between B and C.

**Reply by gaiyf** [reply]
Sorry to bother you again[笑哭] Dr. Xu, There is a problem that was confusing me for a very long time: I used the **TSX**(_Staring Spotlight imaging mode_) data to process, but the result looks incorrect since the third step(**_dem2top_ra.csh_**), I have tried many times but the result always seems erroneous.  The following are some processing files in my experiment. I wonder if the problem is caused by the image or other(my computer's configuration isn't very well).  Looking forward to your reply.

1. the first step in the raw folder:
[TSX20160106.PRM.txt](https://github.com/gmtsar/gmtsar/files/10072381/TSX20160106.PRM.txt)
[TSX20160106 .LED.txt](https://github.com/gmtsar/gmtsar/files/10072382/TSX20160106.LED.txt)

2. the second step in the SLC folder:
[TSX20160106.PRM.txt](https://github.com/gmtsar/gmtsar/files/10072436/TSX20160106.PRM.txt)
[TSX20160106.LED.txt](https://github.com/gmtsar/gmtsar/files/10072439/TSX20160106.LED.txt)

3. the third step in the topo folder
[topo_ra.pdf](https://github.com/gmtsar/gmtsar/files/10072441/topo_ra.pdf)

 4.the forth step in the int folder
 [corr.pdf](https://github.com/gmtsar/gmtsar/files/10072514/corr.pdf)
 [display_amp.pdf](https://github.com/gmtsar/gmtsar/files/10072450/display_amp.pdf)
 [phase.pdf](https://github.com/gmtsar/gmtsar/files/10072476/phase.pdf)
 [phasefilt.pdf](https://github.com/gmtsar/gmtsar/files/10072484/phasefilt.pdf)
 [phasefilt_mask.pdf](https://github.com/gmtsar/gmtsar/files/10072515/phasefilt_mask.pdf)

5.the run record file in shell
[log.txt](https://github.com/gmtsar/gmtsar/files/10072498/log.txt)

**Reply by Xiaohua-Eric-Xu** [reply]
Could it be a PRF problem？Since we only have one pair of test data for TSX development, the recorded PRF may not be the real PRF, you may need to manually change them to (2*PRF or PRF/2) depending on the acquisition mode.

**Reply by gaiyf** [reply]
Thanks very much!
